qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_module.py

```diff
@@ -140,6 +140,20 @@ def parse_dcorr_query_response(response: str) -> _DCORRResponse:
                           secondary=float(secondary))
 
 
+def fixed_negative_float(response: str) -> float:
+    """
+    Keysight sometimes responds for ex. '-0.-1' as an output when you input
+    '-0.1'. This fixed_float function can convert such strings also float.
+    """
+    if len(response.split('.')) != 2:
+        raise Exception('String must of format a.b')
```

Comment:
Are we sure that this will never have to handle an integer string? I could imagine that comming from the keysight too.

Reply:
This should probably return a ValueError to match `float` ?

```
In [3]: float("a")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-a1ff3dbf6f4f> in <module>
----> 1 float("a")

ValueError: could not convert string to float: 'a'
```